include/seqan3/search/configuration/max_error.hpp

```diff
@@ -110,42 +94,40 @@ class max_error : public pipeable_config_element<max_error<errors_t...>, std::ar
      *
      * \include test/snippet/search/configuration_error.cpp
      */
-    constexpr max_error(errors_t && ...errors) noexcept
+    template <typename ...errors_t>
     //!\cond
-        requires sizeof...(errors_t) > 0
+        requires sizeof...(errors_t) > 0 && sizeof...(errors_t) <= 4 &&
+                 ((detail::is_type_specialisation_of_v<std::remove_reference_t<errors_t>, total> ||
+                   detail::is_type_specialisation_of_v<std::remove_reference_t<errors_t>, substitution> ||
+                   detail::is_type_specialisation_of_v<std::remove_reference_t<errors_t>, deletion>  ||
+                   detail::is_type_specialisation_of_v<std::remove_reference_t<errors_t>, insertion>) && ...)
     //!\endcond
-        : base_t{}
+    constexpr max_error(errors_t && ...errors) noexcept : base_t{}
     {
+        static_assert(check_consistency(errors_t{}...), "You may not use the same error specifier more than once.");
+
         detail::for_each([this](auto e)
         {
-            base_t::value[remove_cvref_t<decltype(e)>::_id()] = e.get();
+            switch (remove_cvref_t<decltype(e)>::_id())
+            {
+                case 0 : value.total = e.get(); break;
+                case 1 : value.substitution = e.get(); break;
+                case 2 : value.insertion = e.get(); break;
+                case 3 : value.deletion = e.get(); break;
+            }
         }, std::forward<errors_t>(errors)...);
 
         // Only total is set so we set all other errors to the total limit.
         if constexpr (((std::remove_reference_t<errors_t>::_id() == 0) || ...) && sizeof...(errors) == 1)
         {
-            std::ranges::fill(base_t::value | views::slice(1, 4), base_t::value[0]);
+            value.substitution = value.insertion = value.deletion = value.total;
         } // otherwise if total is not set but any other field is set than use total as the sum of all set errors.
         else if constexpr (!((std::remove_reference_t<errors_t>::_id() == 0) || ...) && sizeof...(errors) > 0)
         {
-            base_t::value[0] = std::min(static_cast<uint8_t>(255), ranges::accumulate(base_t::value | views::slice(1, 4),
-                                                                              static_cast<uint8_t>(0)));
+            value.total = std::min(255u, static_cast<unsigned>(value.substitution) + value.insertion + value.deletion);
```

Comment:
```suggestion
            value.total = std::min<uint32_t>(255, value.substitution + value.insertion + value.deletion);
```

Reply:
I deliberately did this. I see now in CompilerExplorer that it works, but why?!
In your suggestion, the sum (of ins + del+ sub) will be cast into `uint32_t`. Doesn't the cast happen **after** the computation of the sum, but the sum might have overflowed already?

Reply:
No, beacuse there is no `+-operator` for int8_t types defined. So they get promoted to int arithmetics which will cause no overflow in that case.

Reply:
ah, thanks for clarifying